src/BinSkim.Driver/ExpandArguments.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Microsoft.CodeAnalysis.IL
+{
+    public static class ExpandArguments
+    {
+        public static string[] GenerateArguments(
+            string[] args,
+            IFileSystem fileSystem,
+            IEnvironmentVariables environmentVariables)
+        {
+            var expandedArguments = new List<string>();
+
+            foreach (string argument in args)
+            {
+                if (!IsResponseFileArgument(argument))
+                {
+                    expandedArguments.Add(argument);
+                    continue;
+                }
+
+                string responseFile = argument.Trim('"').Substring(1);
+
+                responseFile = environmentVariables.ExpandEnvironmentVariables(responseFile);
+                responseFile = fileSystem.PathGetFullPath(responseFile);
+
+                string[] responseFileLines = fileSystem.FileReadAllLines(responseFile);
+                ExpandResponseFile(responseFileLines, expandedArguments);
+            }
+
+            return expandedArguments.ToArray();
+        }
+
+        private static bool IsResponseFileArgument(string argument)
+        {
+            return argument.Length > 1 && argument[0] == '@';
+        }
+
+        private static void ExpandResponseFile(string[] responseFileLines, List<string> expandedArguments)
```

Comment:
Does this handle `#` for comments ([ref](https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-response-files?view=vs-2022))? - I don't know if it should or whether previous implementation did, but it's something to think about considering we'll also use this functionality for Windows.

Reply:
Great question! I don't believe the original implementation takes that into consideration, but I can see that being an issue. 

I've played with this a bit and yes, It would treat the comments as like a list of items to scan: `"analyze" "/some/path/" "#" "Some" "Random" "Comment"`.
But won't crash, or be valid towards the analysis. But this is something that should be addressed.